packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -121,11 +104,115 @@ class WidthPlatformBreakpoint extends Breakpoint {
 ///
 ///  * [SlotLayout.config], which uses breakpoints to dictate the layout of the
 ///    screen.
-abstract class Breakpoint {
-  /// Returns a const [Breakpoint].
-  const Breakpoint();
+class Breakpoint {
+  /// Returns a const [Breakpoint] with the given constraints.
+  const Breakpoint({
+    this.beginWidth,
+    this.endWidth,
+    this.beginHeight,
+    this.endHeight,
+    this.platform,
+    this.andUp = false,
+  });
+
+  /// Returns a [Breakpoint] with the given constraints for a small screen.
+  const Breakpoint.small({this.andUp = false, this.platform})
+      : beginWidth = 0,
+        endWidth = 600,
+        beginHeight = null,
+        endHeight = 480;
+
+  /// Returns a [Breakpoint] with the given constraints for a medium screen.
+  const Breakpoint.medium({this.andUp = false, this.platform})
+      : beginWidth = 600,
+        endWidth = 840,
+        beginHeight = 480,
+        endHeight = 900;
+
+  /// Returns a [Breakpoint] with the given constraints for a mediumLarge screen.
+  const Breakpoint.mediumLarge({this.andUp = false, this.platform})
+      : beginWidth = 840,
+        endWidth = 1200,
+        beginHeight = 900,
+        endHeight = null;
+
+  /// Returns a [Breakpoint] with the given constraints for a large screen.
+  const Breakpoint.large({this.andUp = false, this.platform})
+      : beginWidth = 1200,
+        endWidth = 1600,
+        beginHeight = 900,
+        endHeight = null;
+
+  /// Returns a [Breakpoint] with the given constraints for an extraLarge screen.
+  const Breakpoint.extraLarge({this.andUp = false, this.platform})
+      : beginWidth = 1600,
+        beginHeight = 900,
+        endWidth = null,
+        endHeight = null;
+
+  /// A set of [TargetPlatform]s that the [Breakpoint] will be active on desktop.
+  static const Set<TargetPlatform> desktop = <TargetPlatform>{
+    TargetPlatform.linux,
+    TargetPlatform.macOS,
+    TargetPlatform.windows
+  };
+
+  /// A set of [TargetPlatform]s that the [Breakpoint] will be active on mobile.
+  static const Set<TargetPlatform> mobile = <TargetPlatform>{
+    TargetPlatform.android,
+    TargetPlatform.fuchsia,
+    TargetPlatform.iOS,
+  };
+
+  /// When set to true, it will include any size above the set width.
+  final bool andUp;
```

Comment:
I wonder if we'll ever need a similar property for `height`. In that case should this be something like `andUpWidth`?

Reply:
Probably not, i can't really think of a use-case. If so we could always add it later if someone opens on issue?